kotskinds/apis/kots/v1beta1/helmchart_types_test.go

```diff
@@ -339,3 +342,183 @@ func Test_MappedChartValueGetValue(t *testing.T) {
+							boolValue: false,
+							valueType: "bool",
+						},
+					},
+				},
+			},
+			overlayValues: map[string]MappedChartValue{
+				"postgres": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"enabled": &MappedChartValue{
+							boolValue: true,
+							valueType: "bool",
+						},
+					},
+				},
+			},
+			expect: map[string]MappedChartValue{
+				"postgres": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"enabled": &MappedChartValue{
+							boolValue: true,
+							valueType: "bool",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "base-only-no-overlay",
+			baseValues: map[string]MappedChartValue{
+				"postgres": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"enabled": &MappedChartValue{
+							boolValue: false,
+							valueType: "bool",
+						},
+					},
+				},
+			},
+			overlayValues: map[string]MappedChartValue{},
+			expect: map[string]MappedChartValue{
+				"postgres": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"enabled": &MappedChartValue{
+							boolValue: false,
+							valueType: "bool",
+						},
+					},
+				},
+			},
+		},
+		{
+			name:       "no-base-only-overlay",
+			baseValues: map[string]MappedChartValue{},
+			overlayValues: map[string]MappedChartValue{
+				"postgres": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"enabled": &MappedChartValue{
+							boolValue: false,
+							valueType: "bool",
+						},
+					},
+				},
+			},
+			expect: map[string]MappedChartValue{
+				"postgres": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"enabled": &MappedChartValue{
+							boolValue: false,
+							valueType: "bool",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "with-deep-children",
+			baseValues: map[string]MappedChartValue{
+				"storage": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"postgres": &MappedChartValue{
+							valueType: "children",
+							children: map[string]*MappedChartValue{
+								"enabled": &MappedChartValue{
+									boolValue: true,
+									valueType: "bool",
+								},
+								"replacementtest": &MappedChartValue{
+									strValue:  "somethinghello",
+									valueType: `string`,
+								},
+							},
+						},
+					},
+				},
+			},
+			overlayValues: map[string]MappedChartValue{
+				"storage": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"postgres": &MappedChartValue{
+							valueType: "children",
+							children: map[string]*MappedChartValue{
+								"enabled": &MappedChartValue{
+									boolValue: true,
+									valueType: "bool",
+								},
+								"replacementtest": &MappedChartValue{
```

Comment:
Isn't the reason for this change when there is a deep merge and the overlay does not contain the key from the base? Or am I missing something? It seems you're missing a test for that case.

Reply:
Nice thanks, added the test case